packages/@aws-cdk/aws-appmesh/lib/http-route-path-match.ts

```diff
@@ -95,3 +95,163 @@ class HttpRouteWholePathMatch extends HttpRoutePathMatch {
+    return new HttpGatewayRoutePrefixPathMatch(prefixPathMatch, prefixRewrite);
+  }
+
+  /**
+   * Returns the gateway route path match configuration.
+   */
+  public abstract bind(scope: Construct): HttpGatewayRoutePathMatchConfig;
+}
+
+class HttpGatewayRoutePrefixPathMatch extends HttpGatewayRoutePathMatch {
+  constructor(
+    private readonly prefixPathMatch: string,
+    private readonly pathRewrite?: HttpGatewayRoutePathRewrite,
+  ) { super(); }
+
+  bind(scope: Construct): HttpGatewayRoutePathMatchConfig {
+    return {
+      prefixPathMatch: this.prefixPathMatch,
+      prefixPathRewrite: this.pathRewrite?.bind(scope).prefixPathRewrite,
+    };
+  }
+}
+
+class HttpGatewayRouteWholePathMatch extends HttpGatewayRoutePathMatch {
+  constructor(
+    private readonly wholePathMatch: CfnGatewayRoute.HttpPathMatchProperty,
+    private readonly wholePathRewrite?: CfnGatewayRoute.HttpGatewayRoutePathRewriteProperty,
+  ) { super(); }
+
+  bind(_scope: Construct): HttpGatewayRoutePathMatchConfig {
+    return {
+      wholePathMatch: this.wholePathMatch,
+      wholePathRewrite: this.wholePathRewrite?.exact
+        ? this.wholePathRewrite
+        : undefined,
+    };
+  }
+}
+
+/**
+ * The type returned from the `bind()` method in {@link HttpGatewayRoutePathRewrite}.
+ */
+export interface HttpGatewayRoutePrefixPathRewriteConfig {
+  /**
+   * Gateway route configuration for rewriting the URL path of the request.
+   *
+   * @default - rewrites the request's URL path to '/'
+   */
+  readonly prefixPathRewrite?: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty;
+}
+
+/**
+ * Defines HTTP gateway route path rewrite based on the URL path of the request.
+ */
+export abstract class HttpGatewayRoutePathRewrite {
+  /**
+   * The default prefix used to replace the incoming route prefix when rewritten.
+   * When enabled, rewrites the matched prefix in Gateway Route to '/'.
+   * When disabled, retains the original prefix from the request.
+   */
+  public static disableDefaultPrefix(): HttpGatewayRoutePathRewrite {
+    return new HttpGatewayRoutePrefixPathRewriteImpl({ defaultPrefix: 'DISABLED' });
+  }
+
+  /**
+   * Replace the incoming route prefix when rewritten.
+   *
+   * @param value the value used to replace the incoming route prefix when rewritten
+   */
+  public static customPrefix(value: string): HttpGatewayRoutePathRewrite {
+    return new HttpGatewayRoutePrefixPathRewriteImpl({ value: value } );
+  }
+
+  /**
+   * Return HTTP gateway route rewrite configuration.
+   */
+  abstract bind(scope: Construct): HttpGatewayRoutePrefixPathRewriteConfig;
+}
+
+class HttpGatewayRoutePrefixPathRewriteImpl extends HttpGatewayRoutePathRewrite {
+  constructor(
+    private readonly prefixRewrite: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty,
+  ) { super(); }
+
+  bind(_scope: Construct): HttpGatewayRoutePrefixPathRewriteConfig {
+    return {
+      prefixPathRewrite: this.prefixRewrite,
+    };
+  }
+}
```

Comment:
If you implement my suggestions from `startsWith()`, all of this can be deleted.

Reply:
Pending

Reply:
Removing entirely